reconcilers/reconcilers_test.go

```diff
@@ -3606,6 +3609,2092 @@ func TestChildReconciler_Unstructured(t *testing.T) {
+			ExpectEvents: []rtesting.Event{
+				rtesting.NewEvent(resource, scheme, corev1.EventTypeWarning, "CreationFailed",
+					"Failed to create ConfigMap %q:  %q already exists", configMapName1, configMapName1),
+			},
+			ExpectCreates: []client.Object{
+				configMapCreate1,
+			},
+		},
+		"child name collision, stale informer cache": {
+			Resource: resourceReady.
+				SpecDie(func(d *dies.TestResourceSpecDie) {
+					d.AddField("foo", "bar")
+				}).
+				DieReleasePtr(),
+			APIGivenObjects: []client.Object{
+				configMapGiven1,
+			},
+			WithReactors: []rtesting.ReactionFunc{
+				rtesting.InduceFailure("create", "ConfigMap", rtesting.InduceFailureOpts{
+					Error: apierrs.NewAlreadyExists(schema.GroupResource{}, configMapName1),
+				}),
+			},
+			Metadata: map[string]interface{}{
+				"SubReconciler": func(t *testing.T, c reconcilers.Config) reconcilers.SubReconciler[*resources.TestResource] {
+					return defaultChildSetReconciler(c)
+				},
+			},
+			ExpectEvents: []rtesting.Event{
+				rtesting.NewEvent(resource, scheme, corev1.EventTypeWarning, "CreationFailed",
+					"Failed to create ConfigMap %q:  %q already exists", configMapName1, configMapName1),
+			},
+			ExpectCreates: []client.Object{
+				configMapCreate1,
+			},
+			ShouldErr: true,
+		},
+		"preserve immutable fields for each child": {
+			Resource: resourceReady.
+				SpecDie(func(d *dies.TestResourceSpecDie) {
+					d.AddField("foo", "bar")
+					d.AddField("qux", "baz")
+				}).
+				StatusDie(func(d *dies.TestResourceStatusDie) {
+					d.AddField("foo", "bar")
+					d.AddField("qux", "baz")
+					d.AddField("immutable", "field")
+				}).
+				DieReleasePtr(),
+			GivenObjects: []client.Object{
+				configMapGiven1.
+					AddData("immutable", "field"),
+				configMapGiven2.
+					AddData("immutable", "field"),
+			},
+			Metadata: map[string]interface{}{
+				"SubReconciler": func(t *testing.T, c reconcilers.Config) reconcilers.SubReconciler[*resources.TestResource] {
+					r := defaultChildSetReconciler(c)
+					r.HarmonizeImmutableFieldsForEachChild = func(current, desired *corev1.ConfigMap) {
+						desired.Data["immutable"] = current.Data["immutable"]
+					}
+					return r
+				},
+			},
+		},
+		"status only reconcile": {
+			Resource: resource.DieReleasePtr(),
+			GivenObjects: []client.Object{
+				configMapGiven1,
+			},
+			ExpectResource: resourceReady.
+				StatusDie(func(d *dies.TestResourceStatusDie) {
+					d.AddField("foo", "bar")
+				}).
+				DieReleasePtr(),
+			Metadata: map[string]interface{}{
+				"SubReconciler": func(t *testing.T, c reconcilers.Config) reconcilers.SubReconciler[*resources.TestResource] {
+					r := defaultChildSetReconciler(c)
+					r.DesiredChildSet = func(ctx context.Context, parent *resources.TestResource) ([]*corev1.ConfigMap, error) {
+						return nil, reconcilers.OnlyReconcileChildStatus
+					}
+					return r
+				},
+			},
+		},
+		"sanitize each child before logging": { //TODO: what's different?
```

Comment:
I copied this test from the `ChildReconciler` test suite but I don't fully understand how we are testing the functionality here. There seems to be no assertion on logs, and changing the return value of `SanitizeForEachChild`, or `SanitizeForChild` in `ChildReconciler` suite, doesn't seem to affect the test runs.